odl/contrib/pytorch/operator.py

```diff
@@ -222,9 +245,162 @@ def backward(self, grad_output):
+            The computed output. Its tensor will have shape
+            ``extra_shape + operator.range.shape``, where ``extra_shape``
+            are the extra axes of ``x``.
+
+        Examples
+        --------
+        Evaluating on a 2D tensor, where the operator expects a 1D input,
+        i.e., with extra batch axis only:
+
+        >>> matrix = np.array([[1, 0, 0],
+        ...                    [0, 1, 1]], dtype='float32')
+        >>> odl_op = odl.MatrixOperator(matrix)
+        >>> odl_op.domain.shape
+        (3,)
+        >>> odl_op.range.shape
+        (2,)
+        >>> op_mod = OperatorAsModule(odl_op)
+        >>> t = torch.ones(3)
+        >>> x = autograd.Variable(t[None, :]) # "fake" batch axis
+        >>> op_mod(x)
+        Variable containing:
+         1  2
+        [torch.FloatTensor of size 1x2]
+        >>> t = torch.ones(3)
+        >>> x_tensor = torch.stack([0 * t, 1 * t])
+        >>> x = autograd.Variable(x_tensor)  # batch of 2 inputs
+        >>> op_mod(x)
+        Variable containing:
+         0  0
+         1  2
+        [torch.FloatTensor of size 2x2]
+
+        An arbitrary number of axes is supported:
+
+        >>> x = autograd.Variable(t[None, None, :])  # "fake" batch and channel
+        >>> op_mod(x)
+        Variable containing:
+        (0 ,.,.) =
+          1  2
+        [torch.FloatTensor of size 1x1x2]
+        >>> x_tensor = torch.stack([torch.stack([0 * t, 1 * t]),
+        ...                         torch.stack([2 * t, 3 * t]),
+        ...                         torch.stack([4 * t, 5 * t])])
+        >>> x = autograd.Variable(x_tensor)  # batch of 3x2 inputs
+        >>> op_mod(x)
+        Variable containing:
+        (0 ,.,.) =
+           0   0
+           1   2
+        <BLANKLINE>
+        (1 ,.,.) =
+           2   4
+           3   6
+        <BLANKLINE>
+        (2 ,.,.) =
+           4   8
+           5  10
+        [torch.FloatTensor of size 3x2x2]
+        """
+        in_shape = x.data.shape
+        op_in_shape = self.op_func.operator.domain.shape
+        op_out_shape = self.op_func.operator.range.shape
+
+        extra_shape = in_shape[:-len(op_in_shape)]
+
+        if in_shape[-len(op_in_shape):] != op_in_shape or not extra_shape:
+            shp_str = str(op_in_shape).strip('()')
+            raise ValueError('expected input of shape (N, *, {}), got input '
+                             'with shape {}'.format(shp_str, in_shape))
+
+        # Flatten extra axes, then do one entry at a time
+        x_flat_xtra = x.view(-1, *op_in_shape)
+        results = []
+        for i in range(x_flat_xtra.data.shape[0]):
+            results.append(self.op_func(x_flat_xtra[i]))
+
+        # Reshape the resulting stack to the expected output shape
+        stack_flat_xtra = torch.stack(results)
+        return stack_flat_xtra.view(extra_shape + op_out_shape)
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        op_name = self.op_func.operator.__class__.__name__
```

Comment:
Why not repr the actual operator?

Reply:
Because of the self-printing of composed modules. For example:
```
NNFBP (
  (conv_det): Conv2d(1, 32, kernel_size=(1, 11), stride=(1, 1), padding=(0, 10))
  (backproj): OperatorAsModule(RayBackProjection) ((50, 256) -> (128, 128))
  (conv1_vol): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(2, 2))
  (conv2_vol): Conv2d(32, 1, kernel_size=(3, 3), stride=(1, 1), padding=(2, 2))
)
```
But I can check if that works with `__str__`.